File: java/src/main/java/com/gmail/fuskerr63/java/interactor/NotificationRepository.java
package com.gmail.fuskerr63.java.interactor;

import com.gmail.fuskerr63.java.entity.BirthdayCalendar;

public interface NotificationRepository {
    void setAlarm(BirthdayCalendar birthdayCalendar, int id, String text, int flag);
    void cancelAlarm(int id, String text, int flag);
    void notifyNotification(int id, String text, int flag, String channelId, int priority);
    boolean alarmIsUp(int id, String text, int flag);
}

File: java/src/main/java/com/gmail/fuskerr63/java/interactor/NotificationRepository.kt
package com.gmail.fuskerr63.java.interactor

import com.gmail.fuskerr63.java.entity.BirthdayCalendar

interface NotificationRepository {
    fun setAlarm(birthdayCalendar: BirthdayCalendar, id: String, text: String)
    fun cancelAlarm(id: String, text: String)
    fun notifyNotification(id: String, text: String, flag: Int, channelId: String, priority: Int)
    fun alarmIsUp(id: String, text: String): Boolean
}
